src/nomad_simulations/outputs.py

```diff
@@ -15,38 +15,153 @@
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
 
-import re
 import numpy as np
+from typing import Optional
+from structlog.stdlib import BoundLogger
 
-from nomad.units import ureg
 from nomad.datamodel.data import ArchiveSection
-from nomad.metainfo import Quantity, SubSection, SectionProxy, MEnum
+from nomad.datamodel.metainfo.annotations import ELNAnnotation
+from nomad.metainfo import Quantity, SubSection, SectionProxy, Reference
+
+from .atoms_state import AtomsState, OrbitalsState
+from .model_system import ModelSystem
+from .numerical_settings import SelfConsistency
 
 
-class Outputs(ArchiveSection):
-    """ """
+class BaseOutputs(ArchiveSection):
+    """
+    Base section to define the output properties of a simulation. This is used as a placeholder for both
+    final `Outputs` properties and the self-consistent (SCF) steps, see `Outputs` base section definition.
+    """
+
+    # TODO add time quantities
 
     normalizer_level = 2
 
+    name = Quantity(
+        type=str,
+        description="""
+        Name of the output property. This is used for easier identification of the property and is conneceted
+        with the class name of each output property class, e.g., `'ElectronicBandGap'`, `'ElectronicBandStructure'`, etc.
+        """,
+        a_eln=ELNAnnotation(component='StringEditQuantity'),
+    )
+
+    orbitals_state_ref = Quantity(
+        type=OrbitalsState,
+        description="""
+        Reference to the `OrbitalsState` section on which the simulation is performed and the
+        output property is calculated.
+        """,
+    )
+
+    atoms_state_ref = Quantity(
+        type=AtomsState,
+        description="""
+        Reference to the `AtomsState` section on which the simulation is performed and the
+        output property is calculated.
+        """,
+    )
+
+    model_system_ref = Quantity(
+        type=ModelSystem,
+        description="""
+        Reference to the `ModelSystem` section on which the simulation is performed and the
+        output property is calculated.
+        """,
+    )
+
+    is_derived = Quantity(
+        type=bool,
+        default=False,
+        description="""
+        Flag indicating whether the output property is derived from other output properties. We make
+        the distinction between directly parsed, derived, and post-processing output properties:
+            - Directly parsed: the output property is directly parsed from the simulation output files.
+            - Derived: the output property is derived from other output properties. No extra numerical settings
+                are required to calculate the output property.
+            - Post-processing: the output property is derived from other output properties. Extra numerical settings
+                are required to calculate the output property.
+        """,
+    )
```

Comment:
1. I think that a generic reference may be appropriate here, or can you guarantee that the option will always be binary?
2. I would not mention `post-processing` here, since it's part of a separate plugin. We can, however, use this comparison in our area-specific docs.

Reply:
Btw, on point 1, this is an interesting concept:
we all agree that FAIR data needs provenance information, but here we're actually logging internal provenance steps.

Reply:
>  can you guarantee that the option will always be binary?

Yes, it is just that we need to clarify that there is a third option. But my argument: let's see how the `nomad-analysis` plugin evolves and how we fit everything altogether.